app/services/target_service/compatible_target_finder.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module TargetService
+  class CompatibleTargetFinder
+    def initialize(target)
+      @target = target
+      @user = target.user
+      @topic = target.topic
+    end
+
+    def call
+      find_compatible_targets
+    end
+
+    private
+
+    def match_targets
+      @match_targets ||= Target.where.not(user: @user).where(topic: @topic)
+    end
+
+    def find_compatible_targets
+      match_targets.each do |match_target|
+        distance = Geocoder::Calculations.distance_between([@target.latitude, @target.longitude],
+                                                           [match_target.latitude, match_target.longitude])
+        if distance <= @target.radius + match_target.radius
```

Comment:
Just as an example, `geokit-rails` provides a [`within`](https://github.com/geokit/geokit-rails#new-scopes-to-use) method that given a point and a radius, it returns targets inside of that circle. Maybe you can find some similar method that does the same out of the box.

Reply:
Thanks @sebastiancaraballo. I already took a look and that is also a good way to calculate this.
Since both solutions work, could I stay with my implementation or should I update the code?

Reply:
The difference is that here this takes more time because it is performing the distance calculation for each possible target. Let's say `match_targets` returns a million targets. For each of them, it calculates a distance. The advantage of using the `within` alternative, is that it only performs one query and doesn't use memory.

Reply:
@sebastiancaraballo of course this is a better way to handle this, so I will update the code. Thanks!